src/TestLogger/Core/InputSanitizerXml.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) Spekt Contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Spekt.TestLogger.Core
+{
+    using System.Text.RegularExpressions;
+
+    public class InputSanitizerXml : IInputSanitizer
+    {
+        public string Sanitize(string input)
+        {
+            if (input == null)
+            {
+                return null;
+            }
+
+            // From xml spec (http://www.w3.org/TR/xml/#charsets) valid chars:
+            // #x9 | #xA | #xD | [#x20-#xD7FF] | [#xE000-#xFFFD] | [#x10000-#x10FFFF]
+
+            // we are handling only #x9 | #xA | #xD | [#x20-#xD7FF] | [#xE000-#xFFFD]
+            // because C# support unicode character in range \u0000 to \uFFFF
+            var evaluator = new MatchEvaluator(ReplaceInvalidCharacterWithUniCodeEscapeSequence);
+            var invalidChar = @"[^\x09\x0A\x0D\x20-\uD7FF\uE000-\uFFFD]";
+            return Regex.Replace(input, invalidChar, evaluator);
```

Comment:
I was wondering if we should consider the new `GeneratedRegex` attribute to create a `Regex` instance at compile time and use it for fast matches. See https://learn.microsoft.com/en-us/dotnet/api/system.text.regularexpressions.generatedregexattribute. More background [here](https://devblogs.microsoft.com/dotnet/regular-expression-improvements-in-dotnet-7/#source-generation).

Our Regex calls are `7 + 2 * NumberOfMessages + 2 * NumberOfTraits`. Say `15` calls per test on average. I think we may benefit from caching with pre-compiled Regex.

Reply:
I'm happy to. But we are on netstandard1.5. Are we stuck with standard for this? I'm not really sure what vstest's requirements are here.

I can at least make this a compiled static regex.

Reply:
I think this pre-compiled regex feature is dependent on the SDK we're using to build. We'll probably need to upgrade the CI yml to .net 7.

Compiled static regex can be less performant than dynamic Regex for smaller number of calls. [This](https://rimdev.io/regex-performance-with-and-without-regexoptions-compiled-using-dotnet-framework-48-and-net-core-31-december-2019/) post goes into some details. Precompiled regex doesn't have the overhead as Compiled and thus may best of both world.

Reply:
We still need the generator to be available as an analyzer for our code, so we have the market attributes available / and to do the build. So it would be more than just the CI, the local build would need the analyzer available or the partial regex methods will break the build.

Reply:
Yes, local build will also need .net 7 sdk installed. Target framework of the project can still remain as `netstandard1.5`.

Reply:
I have tried this a few times and am not clear how this is supposed to work. I can't get it to build locally so I'm stuck, and I have all the SDKs.. Maybe we can break this out to its own PR that will follow this. Seems sensible if you expect we are changing the CI too.

Reply:
Let's separate this to another PR.